dist/bin/kctf-cluster

```diff
@@ -320,11 +320,98 @@ EOF
   echo "${CONFIG_NAME}"
 }
 
+function kctf_cluster_get_ip_ranges {
+  get_cloud_armor_policy || return
+  _kctf_log "Current IP range: \"$ret\""
+}
+
+function kctf_cluster_ip_ranges_usage {
+  echo "usage: kctf cluster set-src-ip-ranges ranges" >&2
+  echo "  -h|--help       print this help" >&2
+  echo "  ranges          ip ranges to allow of the form 1.1.1.1/32,2.2.2.0/24" >&2
+}
+
+function kctf_cluster_ip_ranges {
+  OPTS="h"
+  LONGOPTS="help"
+  PARSED=$(getopt --options=$OPTS --longoptions=$LONGOPTS --name "kctf cluster set-src-ip-ranges" -- "$@")
+  if [[ $? -ne 0 ]]; then
+    kctf_cluster_ip_ranges_usage
+    return 1
+  fi
+  eval set -- "$PARSED"
+
+  while true; do
+    case "$1" in
+      -h|--help)
+        kctf_cluster_ip_ranges_usage
+        return 1
+        ;;
+      --)
+        shift
+        break
+        ;;
+      *)
+        _kctf_log_err "Unrecognized argument \"$1\"."
+        kctf_cluster_ip_ranges_usage
+        return 1
+        ;;
+    esac
+  done
+
+  if [[ $# -ne 1 ]]; then
+    _kctf_log_err "kctf cluster set-src-ip-ranges: expecting a single range argument"
+    kctf_cluster_ip_ranges_usage
+    return 1
+  fi
+
+  RANGES="$1"
+
+  # update the cloud armor policy
+  set_cloud_armor_policy "${RANGES}" || return
+
+  # stop the operator
+  "${KCTF_BIN}/kubectl" delete -f "${KCTF_CTF_DIR}/kctf/resources/operator.yaml" || return
+
+  start_operator_gce || return
+}
+
+function start_operator_gce {
+  get_cloud_armor_policy || return
+  RANGES=$ret
+  # restart the operator with the new range
+  "${KCTF_BIN}/yq" eval "(.spec.template.spec.containers[0].env[] | select(.name == \"ALLOWED_IPS\")).value = \"${RANGES}\"" "${KCTF_CTF_DIR}/kctf/resources/operator.yaml" | "${KCTF_BIN}/kubectl" apply -f - || return
+}
+
+function set_cloud_armor_policy {
+  gcloud compute security-policies rules update --action allow --security-policy kctf-policy --src-ip-ranges "$1" 0 || return
```

Comment:
this probably needs to be namespaced (to support more than 1 cluster per project) like the other kctf stuff

Reply:
I don't think we actually support more than one cluster per project and it makes stuff more complicated.
Do we actually want to support this?

Reply:
in this case, we would need to pass the name to the operator for example

Reply:
#133 fixed the issues we had with multi-cluster support. this means v1.1 doesn't support it (so it's kindof a breaking change)

Reply:
oops, I meant #140

Reply:
that bug is still open though

Reply:
Yes but we only have issues at the challenge name level. It's already possible to have 2 clusters with the same project, we just haven't made the challenge names namespaced too.

If we don't fix this here, then it'll be effectively impossible to have 2 clusters in one project (right now it's just not possible to have 2 tasks with the same name).

Reply:
Using a unique policy name now